src/iidm/util/Networks.cpp

```diff
@@ -28,13 +28,13 @@ stdcxx::CReference<Terminal> getEquivalentTerminal(const VoltageLevel& voltageLe
 
     VoltageLevel::NodeBreakerView::Traverser traverser = [&equivalentTerminal, &voltageLevel](unsigned long /*node1*/, const stdcxx::Reference<Switch>& sw, unsigned long node2) {
         if (sw && sw.get().isOpen()) {
-            return false;
+            return math::TraverseResult::TERMINATE_PATH;
         }
         const auto& terminal = voltageLevel.getNodeBreakerView().getTerminal(node2);
         if (terminal) {
             equivalentTerminal = terminal;
         }
-        return !terminal;
+        return terminal ? math::TraverseResult::TERMINATE_TRAVERSER : math::TraverseResult::CONTINUE;
```

Comment:
With this implementation, you check the value of terminal twice. Maybe this proposal is better:
```
        if (terminal) {
            equivalentTerminal = terminal;
            return math::TraverseResult::TERMINATE_TRAVERSER;
        }
        return math::TraverseResult::CONTINUE;
```